velox/expression/Expr.cpp

```diff
@@ -312,6 +312,21 @@ void Expr::evalSimplifiedImpl(
     }
   }
 
+  // If any errors occurred evaluating the arguments, its possible that the
+  // values for those arguments were not defined which can lead to undefined
```

Comment:
@kgpai @oerling Krishna, turns out we do need this change, but for a different reason. Conjunct special form (AND/OR) assumes that it is invoked only on rows that do not have errors yet. If there are any pre-existing errors on 'rows' passed to Conjunct special form, these will be cleared (unless new errors occur during evaluation of the conjunct inputs). See finalizeErrors in velox/expression/ConjunctExpr.cpp.